pkgs/tools/filesystems/ntfs-3g/default.nix

```diff
@@ -1,30 +1,33 @@
-{ lib, stdenv, fetchurl, pkg-config, mount, libuuid
-, macfuse-stubs, DiskArbitration
+{ lib, stdenv, fetchFromGitHub, autoreconfHook, pkg-config
+, gettext, mount, libuuid, kmod, macfuse-stubs, DiskArbitration
 , crypto ? false, libgcrypt, gnutls
 }:
 
 stdenv.mkDerivation rec {
   pname = "ntfs3g";
-  version = "2021.8.22";
+  version = "2022.5.17";
 
   outputs = [ "out" "dev" "man" "doc" ];
 
-  buildInputs = [ libuuid ] ++ lib.optionals crypto [ gnutls libgcrypt ]
+  buildInputs = [ gettext libuuid ] ++ lib.optionals crypto [ gnutls libgcrypt ]
     ++ lib.optionals stdenv.isDarwin [ macfuse-stubs DiskArbitration ];
-  nativeBuildInputs = [ pkg-config ];
 
-  src = fetchurl {
-    url = "https://tuxera.com/opensource/ntfs-3g_ntfsprogs-${version}.tgz";
-    sha256 = "55b883aa05d94b2ec746ef3966cb41e66bed6db99f22ddd41d1b8b94bb202efb";
+  # Note: libgcrypt is listed here non-optionally because its m4 macros are
+  # being used in ntfs-3g's configure.ac.
+  nativeBuildInputs = [ autoreconfHook libgcrypt pkg-config ];
+
+  src = fetchFromGitHub {
+    owner = "tuxera";
+    repo = "ntfs-3g";
+    rev = version;
+    sha256 = "sha256-xh8cMNIHeJ1rtk5zwOsmcxeedgZ3+MSiWn2UC7y+gtQ=";
   };
 
-  patchPhase = ''
-    substituteInPlace src/Makefile.in --replace /sbin '@sbindir@'
-    substituteInPlace ntfsprogs/Makefile.in --replace /sbin '@sbindir@'
-    substituteInPlace libfuse-lite/mount_util.c \
-      --replace /bin/mount ${mount}/bin/mount \
-      --replace /bin/umount ${mount}/bin/umount
-  '';
+  patches = [
+    # https://github.com/tuxera/ntfs-3g/pull/39
```

Comment:
Can't we fetchpatch those?

Reply:
Currently since they're not merged upstream yet (discussion ongoing...) the only origin for a fetchpatch would be my personal fork, which I think would be overall worse than just carrying these small patches here (my development fork of ntfs-3g isn't meant to be a long term immutable location).

Reply:
Fetching https://github.com/tuxera/ntfs-3g/commit/31d73a11ee7f404c2195a357a9bc97fa2eb48061.patch and https://github.com/tuxera/ntfs-3g/commit/b44a5070d4d6374e61d7ca27c48b620d98b2aa8d.patch is fine.
Vendoring should be avoided to keep the repo size managable.

Reply:
FWIW I strongly disagree -- these commits aren't part of the ntfs-3g repository (not merged) and relying on these URLs would be relying on what is arguably a GitHub bug / implementation detail. The best alternative to vendoring would be fetching from my own GH fork (the authoritative source for these patches, right now) and I don't feel like having my random personal development fork be a stable source of truth for anything.